autopush-common/src/db/bigtable/bigtable_client/mod.rs

```diff
@@ -193,6 +194,85 @@ fn to_string(value: Vec<u8>, name: &str) -> Result<String, DbError> {
     })
 }
 
+pub fn retry_policy(max: usize) -> RetryPolicy {
+    RetryPolicy::default()
+        .with_max_retries(max)
+        .with_jitter(true)
+}
+
+fn retriable_internal_error(status: &RpcStatus) -> bool {
+    match status.code() {
+        RpcStatusCode::UNKNOWN => {
+            "error occurred when fetching oauth2 token" == status.message().to_ascii_lowercase()
+        }
+        RpcStatusCode::INTERNAL => [
+            "rst_stream",
+            "rst stream",
+            "received unexpected eos on data from from server",
+        ]
+        .contains(&status.message().to_lowercase().as_str()),
+        RpcStatusCode::UNAVAILABLE => true,
+        RpcStatusCode::DEADLINE_EXCEEDED => true,
+        _ => false,
+    }
+}
+
+pub fn retryable_describe_table_error(
+    metrics: Arc<StatsdClient>,
+) -> impl Fn(&grpcio::Error) -> bool {
+    move |err| {
+        debug!("🉑 Checking error...{err}");
+        match err {
+            grpcio::Error::RpcFailure(status) => {
+                info!("GRPC Failure :{:?}", status);
+                metrics
```

Comment:
Let's make the metric calls a method (or something along those lines) to avoid all the duplication in this fn.

Reply:
Not really sure it makes that much of a difference, but sure.